oak-core/src/main/java/org/apache/jackrabbit/oak/Oak.java

```diff
@@ -584,6 +584,10 @@ public Oak with(@NotNull Whiteboard whiteboard) {
             LOG.info("Registered Prefetch feature: " + QueryEngineSettings.FT_NAME_PREFETCH_FOR_QUERIES);
             closer.register(prefetchFeature);
             queryEngineSettings.setPrefetchFeature(prefetchFeature);
+            Feature improvedIsNullCostFeature = newFeature(QueryEngineSettings.FT_NAME_IMPROVED_IS_NULL_COST, whiteboard);
```

Comment:
I would argue this being behind a "feature" toggle - it's more of fixing an incorrect behaviour for me.

Reply:
Yes, it is fixing an incorrect behavior. The problem is, someone might depend on the bug! That is the reason to have a feature toggle. See also https://www.hyrumslaw.com/ and https://xkcd.com/1172/